Enables running of functional tests locally

deepakaravindr opened the pull request:
This commit ensures the following
1. Building the NuGet.Clients.sln will restore and build
   GenerateTestPackages as needed
2. It also ensures that all the test extensions, such as, API.Test.dll
   are imported successfully
3. This ensures that nuget.exe from < NuGetGitRoot >.nuget\nuget.exe is
   always used by functional tests as needed. If not present, it will be downloaded
4. This does not ensure that NuGet.Tests.psm1 is imported when launching
   the experimental instance

@yishaigalatzer @danliu @zhili1208  @MeniZalzman @emgarten @feiling

deepakaravindr committed:
This commit ensures the following

1. Building the NuGet.Clients.sln will restore and build
GenerateTestPackages as needed
2. It also ensures that all the test extensions, such as, API.Test.dll
are imported successfully
3. This ensures that nuget.exe from < NuGetGitRoot >\.nuget\nuget.exe is
always used by functional tests as needed. If not present, it will be downloaded
4. This does not ensure that NuGet.Tests.psm1 is imported when launching
the experimental instance

deepakaravindr committed:
Disabling the test and testcases method in A-TopDownloadedPackages.ps1 to ensure simple 'Run-Test' works properly

yishaigalatzer commented:
Can we also import the psm?

Sent from my Windows Phone

---

From: Deepak Aravindakshanmailto:notifications@github.com
Sent: ‎10/‎29/‎2015 7:37 PM
To: NuGet/NuGet.Clientmailto:NuGet.Client@noreply.github.com
Cc: Yishai Galatzermailto:yigalatz@microsoft.com
Subject: [NuGet.Client] Enables running of functional tests locally (#83)

This commit ensures the following
1.  Building the NuGet.Clients.sln will restore and build GenerateTestPackages as needed
2.  It also ensures that all the test extensions, such as, API.Test.dll are imported successfully
3.  This ensures that nuget.exe from < NuGetGitRoot >.nuget\nuget.exe is always used by functional tests as needed. If not present, it will be downloaded
4.  This does not ensure that NuGet.Tests.psm1 is imported when launching the experimental instance

@yishaigalatzerhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fyishaigalatzer&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=UDPuLkqUIM4x6J7PUJxkuxDstmU%2bofZIp2e7S1th8Vs%3d @danliuhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdanliu&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=8DDeezt%2bt%2bs5ikpQtjwonxcoOu%2bY7OgS60o%2b9ZXDus0%3d @zhili1208https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fzhili1208&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=j5EHT54unl0CMGoeHxV8l60jtVt5gSia5gxYF8jV7OY%3d @MeniZalzmanhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMeniZalzman&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=%2b8tV3ioC8TpopGjOuVIovVNlgHoY0J0N7yBkSR1nlWY%3d @emgartenhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2femgarten&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=deBOVdNX8NLgT6oMU%2bg7iggELjoMGiklVM1Al7tnUyA%3d @feilinghttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2ffeiling&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=J1VkdqIZ0EtEEiKCSTcu%2bBqvlHVfNAIMRtj8N%2f3EbmU%3d

---

You can view, comment on, or merge this pull request online at:

  https://github.com/NuGet/NuGet.Client/pull/83https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=qT5lZR2wLXcwa65Ynp0xM8h7wHRXFQgXOIJQlda0Lww%3d

Commit Summary
-   This commit ensures the following

File Changes
-   M NuGet.Clients.slnhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-0&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=2Pqjr2LyDNvBqxBQ44IyJl%2bSfAQHXoVUFXvjtKlFXGw%3d (236)
-   M test/EndToEnd/NuGet.Tests.psm1https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-1&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=MX4DDVJUnCdQuiLAsg%2fN%2bWX48QZDRDvYyDoncQvHnTs%3d (34)
-   A test/TestExtensions/GenerateTestPackages/AssemblySourceFileGenerator.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-2&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=aVz18c%2fXXkcDcRCh4r%2b1%2bzU6M18PLNpEI%2f2nZkGaPpM%3d (351)
-   A test/TestExtensions/GenerateTestPackages/AssemblySourceFileGenerator.tthttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-3&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=KI59YWCP7WVR3QR73Qm0%2fbmn9RJzxHT73bgjahuWGGE%3d (21)
-   A test/TestExtensions/GenerateTestPackages/AssemblySourceFileGenerator_Partial.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-4&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=m%2bZj8741m86EMMXMlqISdTw8BDpgy%2fM6XTgBwIB%2bqJQ%3d (10)
-   A test/TestExtensions/GenerateTestPackages/GenerateTestPackages.csprojhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-5&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=17ubkms2SPav75ukXdEBLYu6KcUoLByEpR5HQHAu4BM%3d (81)
-   A test/TestExtensions/GenerateTestPackages/Program.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-6&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=TQBVyRRU4n4pwSbbVHJLOXn4wSxHcYUMdlVkNgfdPLk%3d (319)
-   A test/TestExtensions/GenerateTestPackages/Properties/AssemblyInfo.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-7&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=MxeEYfgQOaWGmOYo%2fL589DA%2borUM00Pv54NWArB%2fKZk%3d (7)
-   A test/TestExtensions/GenerateTestPackages/Test.dgmlhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-8&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=UrM0%2fqCD8TqCO%2bl1GCof1f%2bvxU%2b3O38kD2HEjGzLQL8%3d (37)
-   A test/TestExtensions/GenerateTestPackages/TestPackageKey.snkhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-9&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=xqj6uV%2bED1VT5TRw3dwwmrZFh5ZO%2fhCrQ6v9yTLFn3U%3d (0)
-   A test/TestExtensions/GenerateTestPackages/app.confighttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-10&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=kYc57cGT9xf7JKhARxlwcYdEO4vpBXVtRZPeUsgruhQ%3d (3)
-   A test/TestExtensions/GenerateTestPackages/project.jsonhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-11&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ik36hoDjUb2rI6JhrzRzOGv%2b9%2f4u2Hd0muNqlT%2fF4Y8%3d (11)

Patch Links:
-   https://github.com/NuGet/NuGet.Client/pull/83.patchhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83.patch&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=SCKrFA4Ounrw%2bwpbSrXI48Abd5asDdjUGkBA%2bK4Wn6Q%3d
-   https://github.com/NuGet/NuGet.Client/pull/83.diffhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83.diff&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=d6GzX8ucltqfpIvuSW6B4FH%2fALPvTjaaQqtdyItr%2fyw%3d

—
Reply to this email directly or view it on GitHubhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=qT5lZR2wLXcwa65Ynp0xM8h7wHRXFQgXOIJQlda0Lww%3d.

deepakaravindr commented:
I will do that. I was not sure how to achieve this when I sent the PR. One idea I have is to set the environment variable to load the psm1 from build.ps1. So, if build.ps1 was run at least once, it will inport the psm1

Thanks,
Deepak

---

From: Yishai Galatzer notifications@github.com
Sent: Thursday, October 29, 2015 8:16:04 PM
To: NuGet/NuGet.Client
Cc: Deepak Aravindakshan
Subject: Re: [NuGet.Client] Enables running of functional tests locally (#83)

Can we also import the psm?

Sent from my Windows Phone

---

From: Deepak Aravindakshanmailto:notifications@github.com
Sent: ‎10/‎29/‎2015 7:37 PM
To: NuGet/NuGet.Clientmailto:NuGet.Client@noreply.github.com
Cc: Yishai Galatzermailto:yigalatz@microsoft.com
Subject: [NuGet.Client] Enables running of functional tests locally (#83)

This commit ensures the following
1. Building the NuGet.Clients.sln will restore and build GenerateTestPackages as needed
2. It also ensures that all the test extensions, such as, API.Test.dll are imported successfully
3. This ensures that nuget.exe from < NuGetGitRoot >.nuget\nuget.exe is always used by functional tests as needed. If not present, it will be downloaded
4. This does not ensure that NuGet.Tests.psm1 is imported when launching the experimental instance

@yishaigalatzerhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fyishaigalatzer&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=UDPuLkqUIM4x6J7PUJxkuxDstmU%2bofZIp2e7S1th8Vs%3d @danliuhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdanliu&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=8DDeezt%2bt%2bs5ikpQtjwonxcoOu%2bY7OgS60o%2b9ZXDus0%3d @zhili1208https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fzhili1208&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=j5EHT54unl0CMGoeHxV8l60jtVt5gSia5gxYF8jV7OY%3d @MeniZalzmanhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMeniZalzman&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=%2b8tV3ioC8TpopGjOuVIovVNlgHoY0J0N7yBkSR1nlWY%3d @emgartenhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2femgarten&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=deBOVdNX8NLgT6oMU%2bg7iggELjoMGiklVM1Al7tnUyA%3d @feilinghttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2ffeiling&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=J1VkdqIZ0EtEEiKCSTcu%2bBqvlHVfNAIMRtj8N%2f3EbmU%3d

---

You can view, comment on, or merge this pull request online at:

https://github.com/NuGet/NuGet.Client/pull/83https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=qT5lZR2wLXcwa65Ynp0xM8h7wHRXFQgXOIJQlda0Lww%3d

Commit Summary
- This commit ensures the following

File Changes
- M NuGet.Clients.slnhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-0&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=2Pqjr2LyDNvBqxBQ44IyJl%2bSfAQHXoVUFXvjtKlFXGw%3d (236)
- M test/EndToEnd/NuGet.Tests.psm1https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-1&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=MX4DDVJUnCdQuiLAsg%2fN%2bWX48QZDRDvYyDoncQvHnTs%3d (34)
- A test/TestExtensions/GenerateTestPackages/AssemblySourceFileGenerator.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-2&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=aVz18c%2fXXkcDcRCh4r%2b1%2bzU6M18PLNpEI%2f2nZkGaPpM%3d (351)
- A test/TestExtensions/GenerateTestPackages/AssemblySourceFileGenerator.tthttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-3&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=KI59YWCP7WVR3QR73Qm0%2fbmn9RJzxHT73bgjahuWGGE%3d (21)
- A test/TestExtensions/GenerateTestPackages/AssemblySourceFileGenerator_Partial.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-4&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=m%2bZj8741m86EMMXMlqISdTw8BDpgy%2fM6XTgBwIB%2bqJQ%3d (10)
- A test/TestExtensions/GenerateTestPackages/GenerateTestPackages.csprojhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-5&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=17ubkms2SPav75ukXdEBLYu6KcUoLByEpR5HQHAu4BM%3d (81)
- A test/TestExtensions/GenerateTestPackages/Program.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-6&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=TQBVyRRU4n4pwSbbVHJLOXn4wSxHcYUMdlVkNgfdPLk%3d (319)
- A test/TestExtensions/GenerateTestPackages/Properties/AssemblyInfo.cshttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-7&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=MxeEYfgQOaWGmOYo%2fL589DA%2borUM00Pv54NWArB%2fKZk%3d (7)
- A test/TestExtensions/GenerateTestPackages/Test.dgmlhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-8&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=UrM0%2fqCD8TqCO%2bl1GCof1f%2bvxU%2b3O38kD2HEjGzLQL8%3d (37)
- A test/TestExtensions/GenerateTestPackages/TestPackageKey.snkhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-9&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=xqj6uV%2bED1VT5TRw3dwwmrZFh5ZO%2fhCrQ6v9yTLFn3U%3d (0)
- A test/TestExtensions/GenerateTestPackages/app.confighttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-10&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=kYc57cGT9xf7JKhARxlwcYdEO4vpBXVtRZPeUsgruhQ%3d (3)
- A test/TestExtensions/GenerateTestPackages/project.jsonhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%2ffiles%23diff-11&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ik36hoDjUb2rI6JhrzRzOGv%2b9%2f4u2Hd0muNqlT%2fF4Y8%3d (11)

Patch Links:
- https://github.com/NuGet/NuGet.Client/pull/83.patchhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83.patch&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=SCKrFA4Ounrw%2bwpbSrXI48Abd5asDdjUGkBA%2bK4Wn6Q%3d
- https://github.com/NuGet/NuGet.Client/pull/83.diffhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83.diff&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=d6GzX8ucltqfpIvuSW6B4FH%2fALPvTjaaQqtdyItr%2fyw%3d

—
Reply to this email directly or view it on GitHubhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83&data=01%7c01%7cyigalatz%40microsoft.com%7cf487cebbf3bd434d6a4708d2e0d2fd4b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=qT5lZR2wLXcwa65Ynp0xM8h7wHRXFQgXOIJQlda0Lww%3d.

—
Reply to this email directly or view it on GitHubhttps://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.Client%2fpull%2f83%23issuecomment-152400312&data=01%7c01%7cdaravind%40microsoft.com%7c5c2108b1fcfb4799397008d2e0d8732c%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=Vd0213oZkapCkCjf4NBgl3hZokIEk4UBTr8fMM4rq6k%3d.

deepakaravindr committed:
Updated the Platform for GenerateTestPackages to be 'AnyCPU' to match
the platforms of other projects in NuGet.Clients.sln

deepakaravindr committed:
Removed all the accidentally added entries in the solution for
platform 'x86'

deepakaravindr committed:
Setting up functional tests to set the environment variable,
and registry keys.

emgarten commented on SetupFunctionalTests.ps1:
```diff
@@ -0,0 +1,34 @@
+}
+
+# Set the registry key to prevent the 'Not Responding' window from showing up on a crash
+Set-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows\Windows Error Reporting" -Name ForceQueue -Value 1
```
Let's not change this setting. If the CI needs it that is fine, but I wouldn't like changing this on my dev box.

MeniZalzman replied:
+1, we should not things like that because it will hide issues that we want to surface while running the functional tests

emgarten commented:
This will be so much better. Remove the crash setting dialog keys and then :shipit:

deepakaravindr committed:
1. Removed the registry key setting to avoid crash dialog
2. Fixed the setup functional tests to also handle security message for
Text Templating. It is another registry key setting

deepakaravindr commented:
![image](https://cloud.githubusercontent.com/assets/2380340/11107466/fd12b768-8892-11e5-87cf-c0332fd1e137.png)

feiling commented:
Looks great! :shipit:

yishaigalatzer commented:
:shipit:

deepakaravindr commented:
Merged with commits from https://github.com/NuGet/NuGet.Client/commit/327eb72d2861b37a99dff9132ae2721c311489b6 to https://github.com/NuGet/NuGet.Client/commit/224c956f6f22abe884e5b2254be3fe6d5a823bc4
